lib/primer/classify.rb

```diff
@@ -152,98 +159,102 @@ def validated_class_names(classes)
-            elsif BORDER_MARGIN_KEYS.include?(key)
-              memo[:classes] << "#{key.to_s.dasherize}-#{val}"
-            elsif key == DIRECTION_KEY
-              memo[:classes] << "flex#{breakpoint}-#{dasherized_val}"
-            elsif key == JUSTIFY_CONTENT_KEY
-              formatted_value = val.to_s.gsub(/(flex\_|space\_)/, "")
-              memo[:classes] << "flex#{breakpoint}-justify-#{formatted_value}"
-            elsif key == ALIGN_ITEMS_KEY
-              memo[:classes] << "flex#{breakpoint}-items-#{val.to_s.gsub("flex_", "")}"
-            elsif key == FLEX_KEY
-              memo[:classes] << "flex-#{val}"
-            elsif key == FLEX_GROW_KEY
-              memo[:classes] << "flex-grow-#{val}"
-            elsif key == FLEX_SHRINK_KEY
-              memo[:classes] << "flex-shrink-#{val}"
-            elsif key == ALIGN_SELF_KEY
-              memo[:classes] << "flex-self-#{val}"
-            elsif key == WIDTH_KEY || key == HEIGHT_KEY
-              if val == :fit || val == :fill
-                memo[:classes] << "#{key}-#{val}"
-              else
-                memo[key] = val
-              end
-            elsif TEXT_KEYS.include?(key)
-              memo[:classes] << "text-#{dasherized_val}"
-            elsif TYPOGRAPHY_KEYS.include?(key)
-              memo[:classes] << "f#{dasherized_val}"
-            elsif MARGIN_DIRECTION_KEYS.include?(key) && val < 0
-              memo[:classes] << "#{key.to_s.dasherize}#{breakpoint}-n#{val.abs}"
-            elsif key == BOX_SHADOW_KEY
-              if val == true
-                memo[:classes] << "box-shadow"
-              else
-                memo[:classes] << "box-shadow-#{dasherized_val}"
-              end
-            elsif key == VISIBILITY_KEY
-              memo[:classes] << "v-#{dasherized_val}"
-            else
-              memo[:classes] << "#{key.to_s.dasherize}#{breakpoint}-#{dasherized_val}"
-            end
+        memo
+      end
+
+      def extract_value(memo, key, val, breakpoint)
+        return if val.nil?
+        if SPACING_KEYS.include?(key)
+          if MARGIN_DIRECTION_KEYS.include?(key)
+            raise ArgumentError, "value of #{key} must be between -6 and 6" if (val < -6 || val > 6)
+          elsif !((key == :mx || key == :my) && val == :auto)
+            raise ArgumentError, "value of #{key} must be between 0 and 6" if (val < 0 || val > 6)
           end
         end
 
-        {
-          classes: out[:classes].join(" "),
-          styles: out[:styles].join(" ")
-        }.merge(out.except(:classes, :styles))
+        if BOOLEAN_MAPPINGS.has_key?(key)
+          BOOLEAN_MAPPINGS[key][:mappings].map { |m| m[:css_class] if m[:value] == val }.compact.each do |css_class|
+            memo[:classes] << css_class
+          end
+        elsif key == BG_KEY
+          if val.to_s.starts_with?("#")
+            memo[:styles] << "background-color: #{val};"
+          else
+            memo[:classes] << "bg-#{val.to_s.dasherize}"
```

Comment:
should we have a helper:
```rb
def dasherize(val)
  val.to_s.dasherize
end
```
?

Reply:
Good question, I'm leaning slightly towards keeping it as-is, since it's calling `to_s.dasherize` and not doing a more complex operation.